crates/subspace-wasm-tools/src/lib.rs

```diff
@@ -103,3 +109,57 @@ pub fn read_core_domain_runtime_blob(
 
     Ok(embedded_runtime_blob.to_vec())
 }
+
+struct RuntimeCodeFetcher<'a> {
```

Comment:
I'm not sure this is a good place for such code. This is crate with tools for building WASM modules (see crate description). While added logic is basically runtime handling logic.

Reply:
not sure I agree with that though I do i agree that description needs to change. The added functionality is fairly generic that IMO should belong to WASM tools or utils crate.